receiver/couchdbreceiver/client.go

```diff
@@ -0,0 +1,139 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package couchdbreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/couchdbreceiver"
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+)
+
+const nodeNamesPath = "/_membership"
+
+// client defines the basic HTTP client interface.
+type client interface {
+	Get(path string) ([]byte, error)
+	GetNodeNames() ([]string, error)
+	GetStats(nodeName string) (map[string]interface{}, error)
+}
+
+var _ client = (*couchDBClient)(nil)
+
+type couchDBClient struct {
+	client *http.Client
+	cfg    *Config
+	logger *zap.Logger
+}
+
+// newCouchDBClient creates a new client to make requests for the CouchDB receiver.
+func newCouchDBClient(cfg *Config, host component.Host, logger *zap.Logger) (client, error) {
+	client, err := cfg.ToClient(host.GetExtensions())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create HTTP Client: %w", err)
+	}
+
+	return &couchDBClient{
+		client: client,
+		cfg:    cfg,
+		logger: logger,
+	}, nil
+}
+
+// Get issues an authorized Get requests to the specified url.
+func (c *couchDBClient) Get(path string) ([]byte, error) {
+	req, err := c.buildReq(path)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		if err = resp.Body.Close(); err != nil {
+			c.logger.Warn("failed to close response body", zap.Error(err))
+		}
+	}()
+
+	if resp.StatusCode != http.StatusOK {
+		if resp.StatusCode >= 400 {
+			c.logger.Error("couchdb", zap.Error(err), zap.String("status_code", strconv.Itoa(resp.StatusCode)))
+		}
+		return nil, fmt.Errorf("request GET %s failed - %q", req.URL.String(), resp.Status)
+	}
+
+	body, err := io.ReadAll(resp.Body)
```

Comment:
Is the body always going to be within a reasonable limit?

Reply:
Through testing the response size it looks to be a reasonable amount through testing. What is your concern or is there a limit we should be following?

Reply:
If you expect this to be in the megabytes, you might want to return a buffer reader instead of `[]byte`